protocol-units/da/m1/light-node/src/v1/sequencer.rs

```diff
@@ -190,30 +212,16 @@ impl LightNodeV1 {
 		
 		// get some blocks in a batch
 		let blocks = self.read_blocks(receiver).await?;
-
-		// form blobs from the blocks
-		info!(
-			target: "movement_timing",
-			batch_size = blocks.len(),
-			"building_block_batch"
-		);
-		let mut block_blobs = Vec::new();
-		let mut ids = Vec::new();
-		for block in &blocks {
-			let block_blob = self.pass_through.create_new_celestia_blob(
-				serde_json::to_vec(&block)
-					.map_err(|e| anyhow::anyhow!("Failed to serialize block: {}", e))?,
-			)?;
-			block_blobs.push(block_blob);
-			ids.push(block.id());
+		if blocks.is_empty() {
+			return Ok(());
 		}
-
+		let ids = blocks.iter().map(|b| b.id()).collect::<Vec<_>>();
 
 		// submit the blobs, resizing as needed
 		for block_id in &ids {
 			info!(target: "movement_timing", %block_id, "submitting_block_batch");
 		}
```

Comment:
Why allocate a vector just for logging when you can log while iterating over blocks themselves?
```suggestion
		// submit the blobs, resizing as needed
		for block_id in blocks.iter().map(|b| b.id()) {
			info!(target: "movement_timing", %block_id, "submitting_block_batch");
		}
```

Reply:
This is for timing the entire heuristic.